traffic_ops/testing/api/utils/utils.go

```diff
@@ -73,3 +76,58 @@ func Compare(t *testing.T, expected []string, alertsStrs []string) {
 		}
 	}
 }
+
+type CkReqFunc func(*testing.T, toclientlib.ReqInf, interface{}, error)
+
+func CkRequest(c ...CkReqFunc) []CkReqFunc {
+	return c
+}
+
+func NoError() CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, _ interface{}, err error) {
+		assert.NoError(t, err, "Expected no error, but got %v", err)
+	}
+}
+
+func HasError() CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, resp interface{}, err error) {
+		switch resp := resp.(type) {
+		case tc.CacheGroupDetailResponse:
+			assert.Error(t, err, "Expected error - but got: %+v %+v", resp.Response, resp.Alerts)
+		case tc.CacheGroupsNullableResponse:
+			assert.Error(t, err, "Expected error - but got: %+v %+v", resp.Response, resp.Alerts)
+		case tc.Alerts:
+			assert.Error(t, err, "Expected error - but got: %+v", resp.Alerts)
+		default:
+			assert.Fail(t, "Failed to find response type")
+		}
+	}
+}
```

Comment:
If this switch has to encompass all types returned by the API it's going to get extremely long and annoying to maintain. Instead, I'd suggest just taking in the Response and the Alerts as separate arguments.

Ideally, though, the Alerts should just be sufficient to explain why a request failed, and in that case the Response is always supposed to be empty anyway (as far as I can remember right now). If they don't, then it's either a server-side error or a problem that needs to be fixed in the API, IMO. I think seeing response payloads might just make the output difficult to read without providing any real value.

Reply:
`assert.Error` and `assert.Fail` are not much different in implementation, at the core they call `t.Errorf`, which does not terminate the test execution like `FailNow`. Fail simply provides a way to fail without performing any checks, and in this case the checks are performed above. However, this will probably be changed since this library will probably get removed.